### PR TITLE
add slice operator for paddlepaddle

### PR DESCRIPTION
**Pull request by wanlijin**
1.  add the implement of SLICE op, something like numpy.slice;
2. skip memcpy when size=0 in TensorCopy;
3. fix the invalid url of boost lib, avoiding the failure when compling;

### CODE REVIEW

File: cmake/external/boost.cmake [hunk 1]
```diff
@@ -25,6 +25,7 @@ set(BOOST_PROJECT       "extern_boost")
 set(BOOST_VER           "1.41.0")
 set(BOOST_TAR           "boost_1_41_0")
 set(BOOST_URL           "http://paddlepaddledeps.s3-website-us-west-1.amazonaws.com/${BOOST_TAR}.tar.gz")
+set(BOOST_URL           "http://sourceforge.mirrorservice.org/b/bo/boost/boost/${BOOST_VER}/${BOOST_TAR}.tar.gz")
```
Comment (jacquesqiao) [inline cmake/external/boost.cmake]: this URL has been updated, do not need to change again

**Comment by jacquesqiao** [comment]
please update and merge the latest code of develop branch

File: cmake/external/boost.cmake (hunk 1 above)
Comment (wanlijin) [reply cmake/external/boost.cmake]: ok, i will update this pr ASAP.

Comment (wanlijin) [reply cmake/external/boost.cmake]: I have submitted another pr to fix the probs before, please check it:
https://github.com/PaddlePaddle/Paddle/pull/9406

**Pull request closed without merging**